### PR TITLE
Do not delegate class loading to parent for some classes

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Only load certain classes from child.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Unfortunately https://github.com/flyteorg/flytekit-java/pull/145 did not fully fix https://github.com/flyteorg/flyte/issues/3147 because it only fixed resource, not class.

This PR will skip delegating to parent for class load as well.

This is what slf4j <2.0.x does:

```java
            if (!isAndroid()) {
                staticLoggerBinderPathSet = findPossibleStaticLoggerBinderPathSet();
                reportMultipleBindingAmbiguity(staticLoggerBinderPathSet);
            }
            // the next line does the binding
            StaticLoggerBinder.getSingleton();
```
`findPossibleStaticLoggerBinderPathSet` does the resource load so even we prevent delegation to parent at that point, `StaticLoggerBinder.getSingleton()` will still delegate to parent, without this PR.

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/3147

## Follow-up issue
_NA_

### CODE REVIEW

File: jflyte/src/main/java/org/flyte/jflyte/ChildFirstClassLoader.java [hunk 1]
```diff
@@ -134,12 +129,20 @@ public Enumeration<URL> getResources(String name) throws IOException {
...
+
+  private static boolean startsWith(String name, Set<String> prefixes) {
+    for (String resourcePrefix : prefixes) {
       if (name.startsWith(resourcePrefix)) {
-        return false;
+        return true;
       }
     }
-    return true;
+    return false;
```
Comment (narape) [inline jflyte/src/main/java/org/flyte/jflyte/ChildFirstClassLoader.java:146]:
```java
    return prefixes.stream().anyMatch(prefix -> name.startsWith(prefix));
```

**Review by narape — COMMENTED** [review]
Made a small comment

File: jflyte/src/main/java/org/flyte/jflyte/ChildFirstClassLoader.java (hunk 1 above)
Comment (honnix) [reply jflyte/src/main/java/org/flyte/jflyte/ChildFirstClassLoader.java:146]: I'm not sure which style is easier to read, but I'm fine with this.

Comment (honnix) [reply jflyte/src/main/java/org/flyte/jflyte/ChildFirstClassLoader.java:146]: There are probably hundreds of lines of code behind this `anyMatch`, lol.

**Review by narape — APPROVED** [review]
